client/my-sites/upgrades/components/form/state-select.jsx

```diff
@@ -1,32 +1,32 @@
 /**
  * External dependencies
  */
-const React = require( 'react' ),
-	classNames = require( 'classnames' ),
-	isEmpty = require( 'lodash/isEmpty' ),
-	ReactDom = require( 'react-dom' ),
-	observe = require( 'lib/mixins/data-observe' );
+import React, { Component, PropTypes } from 'react';
+import classNames from 'classnames';
+import { connect } from 'react-redux';
+import isEmpty from 'lodash/isEmpty';
+import { localize } from 'i18n-calypso';
+import ReactDom from 'react-dom';
 
 /**
  * Internal dependencies
  */
-const analytics = require( 'lib/analytics' ),
-	FormLabel = require( 'components/forms/form-label' ),
-	FormSelect = require( 'components/forms/form-select' ),
-	FormInputValidation = require( 'components/forms/form-input-validation' ),
-	scrollIntoViewport = require( 'lib/scroll-into-viewport' ),
-	Input = require( './input' );
+import analytics from 'lib/analytics';
+import FormLabel from 'components/forms/form-label';
+import FormSelect from 'components/forms/form-select';
+import FormInputValidation from 'components/forms/form-input-validation';
+import { getCountryStates } from 'state/country-states/selectors';
+import Input from './input';
+import QueryCountryStates from 'components/data/query-country-states';
+import scrollIntoViewport from 'lib/scroll-into-viewport';
 
-module.exports = React.createClass( {
-	displayName: 'StateSelect',
+class StateSelect extends Component {
 
-	mixins: [ observe( 'statesList' ) ],
-
-	recordStateSelectClick: function() {
+	recordStateSelectClick() {
 		if ( this.props.eventFormName ) {
 			analytics.ga.recordEvent( 'Upgrades', `Clicked ${ this.props.eventFormName } State Select` );
```

Comment:
Since this is a connected component, let's use the [recordGoogleEvent](https://github.com/Automattic/wp-calypso/blob/master/client/state/analytics/actions.js#L55) redux action . This makes it a lot easier to write any component tests.

So:
```javascript
analytics.ga.recordEvent( 'Upgrades', `Clicked ${ this.props.eventFormName } State Select` );
```

turns into
```javascript
const { recordGoogleEvent } = this.props;
recordGoogleEvent( 'Upgrades', `Clicked ${ this.props.eventFormName } State Select` );
```